knowledge-base/githubocto/repo-visualizer/action-security.yml

```diff
@@ -0,0 +1,17 @@
+name: 'Repo Visualizer'
+github-token:
+  action-input:
+    input: github_token
+    is-default: true
+  permissions:
+    contents: read
+    contents-reason: to read repo directories
+    pull-requests: write
+    pull-requests-reason: to push, commit and update diagram into the repo
```

Comment:
@arjundashrath can you please add a link to the reason, where it has a reason the contents-read is needed and pull-requests-reason is needed. That makes it easy for me to verify this. You can add it as a comment. See [this](https://github.com/step-security/secure-workflows/blob/8e3478a5c48e4e82ab08272884d0051785095c71/knowledge-base/Azure/static-web-apps-deploy/action-security.yml#L4) for example.

Reply:
```
permissions:
  contents: write
  contents-write-reason: to push the diagram # https://github.com/githubocto/repo-visualizer/blob/main/README.md#accessing-the-diagram
  contents-write-if: {{ not contains should_push || should_push is set to true }}
```